src/internal/connector/discovery/discovery.go

```diff
@@ -43,6 +45,10 @@ func Users(ctx context.Context, ga getAller, errs *fault.Errors) ([]models.Usera
 func User(ctx context.Context, gwi getWithInfoer, userID string) (models.Userable, *api.UserInfo, error) {
 	u, err := gwi.GetByID(ctx, userID)
 	if err != nil {
+		if graph.IsErrUserNotFound(err) {
+			return nil, nil, fmt.Errorf("resource owner [%s] not found within tenant", userID)
```

Comment:
Nit: time to get in the habit of using clues
```suggestion
			return nil, nil, clues.Wrap(err, "resource owner not found within tenant").With("resource_owner", userID)
```

Reply:
with the suggested change the output on the CLI when corso is run with a user that does exist is
```text
Error: 1 error occurred:
        * Failed to run Exchange backup for user foo@other.onmicrosoft.com: doing backup: producing backup data collections: resource owner not found within tenant: getting user: maximum retries or unretryable:  Request_ResourceNotFound Resource 'foo@other.onmicrosoft.com' does not exist or one of its queried reference-property objects are not present.: error status code received from the API
```

Is this something we're comfortable with?

Reply:
Me?  Yes.  Part of this migration is to have a centralized source for error interpretation; and that's still forthcoming.  But it does mean we'll improve in the future.  In the meantime, even though this is unsightly, it's understandable.

@gmatev might not share my opinion.